src/parser/jobs/mnk/modules/Gauge.ts

```diff
@@ -0,0 +1,129 @@
+import {t} from '@lingui/macro'
+
+import ACTIONS from 'data/ACTIONS'
+import JOBS from 'data/JOBS'
+import STATUSES from 'data/STATUSES'
+import {BuffEvent, CastEvent} from 'fflogs'
+
+import {dependency} from 'parser/core/Module'
+import Combatants from 'parser/core/modules/Combatants'
+import {CounterGauge, Gauge as CoreGauge, TimerGauge} from 'parser/core/modules/Gauge'
+
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+import {FORMS} from './Forms'
+
+// GL stack caps.
+export const MAX_STACKS = 3
+export const MAX_FASTER = 4
+
+// Actions that grant stacks.
+const GL_GRANTERS: number[] = [
+	ACTIONS.DEMOLISH.id,
+	ACTIONS.ROCKBREAKER.id,
+	ACTIONS.SNAP_PUNCH.id,
+]
+
+// Actions that refresh the timer without granting stacks.
+const GL_REFRESHERS: number[] = [
+	// kinda weird, only when going Coeurl->Opo-Opo and you need at least one stack
+	ACTIONS.FORM_SHIFT.id,
+	ACTIONS.SIX_SIDED_STAR.id,
+]
+
+// The duration for timeouts
+const GL_TIMEOUT_MILLIS = STATUSES.GREASED_LIGHTNING.duration * 1000
+
+export default class Gauge extends CoreGauge {
+	static handle = 'greasedLightning'
+	static title = t('mnk.gauge.title')`Greased Lightning`
+	static displayOrder = DISPLAY_ORDER.GREASED_LIGHTNING
+
+	@dependency private combatants!: Combatants
+
+	private _stacks = this.add(new CounterGauge({
+		maximum: MAX_STACKS,
+		chart: {label: 'Stacks', color: JOBS.MONK.colour},
+	}))
+
+	private _timer = this.add(new TimerGauge({
+		maximum: GL_TIMEOUT_MILLIS,
+		chart: {label: 'Duration', color: JOBS.PALADIN.colour},
+		onExpiration: () => this._stacks.reset(),
+	}))
+
+	protected init(): void {
+		super.init()
+
+		this.addHook('applybuff', {to: 'player', abilityId: STATUSES.FISTS_OF_WIND.id}, this.goFast)
```

Comment:
you get a ![](https://ffxiv.gamerescape.com/w/images/2/2f/Slap_Emote_Icon.png) for all of these callback names tbqh